202204-datestimesR/notes.Rmd

```diff
@@ -0,0 +1,272 @@
+ - if character string is <i>not</i> already in default format, the `format` argument should be specified to avoid ambiguity
+  - there is an optional timzezone or `tz` argument - more on this later
+ 
+|Type |Code   |Value |
+|---|---|---|
+|Day |%j | Day of year (1-366) |
+|Day |%d | Day of the month (1-31) |
+|Month |%m | Month (01-12) |
+|Month |%b | Month(abbreviated) |
+|Month |%B | Month(full name) |
+|Year |%y | Year (2 digit) |
+|Year |%Y | Year (4 digit) |
+ 
+Helpful hint: If you can't recall the appropriate code, type `?strptime` into your R console and scroll down.  
+
+Let's explore the following:
+```{r, results='hide'}
+x <- as.Date("2021-02-23")
+x
+class(x)
+
+y <- as.Date("April 2, 2019", format = "%B %d, %Y")
+y
+
+#Try converting the following: 
+"Sep 02 1995"
+"19920613"
+"12-Mar-12"
+"2019 156"
+```
+
+#### 2. Date/time representation in POSIX
+
+<i>Content based on [Berkeley Stats curriculum page](https://www.stat.berkeley.edu/~s133/dates.html)</i>
+
+ - POSIX stands for '<b>P</b>ortable <b>O</b>perating <b>S</b>ystem <b>I</b>nterface', primarily for UNI<b>X</b>
+ - POSIXlt stores a list of elements for second, minute, hour, day, month, and year
+ - POSIXct stores date/time as number of seconds since January 1, 1970
+    - POSIXct is the standard choice for dates and times in R
+    - POSIX can also be used to store dates, or date/times without seconds
+ - to convert character string to POSIXct, use `as.POSIXct()`
+   - default ISO 8601 standard is "YYYY-MM-DD hh:mm:ss"
+   - if character string is <i>not</i> already in default format, the `format` argument should be specified to avoid ambiguity
+   - if timezone is not specified, `as.POSIXct()` will default to your `Sys.time()` timezone
+ - to convert components of date/time into POSIXct, use `ISOdate()` or `ISOdatetime()`
+   - specify numeric value of year, month, day, hour, min, sec
+   - if timezone is not specified, `ISOdate()` will default to GMT
+   - if timezone is not specified, `ISOdatetime()` will default to your `Sys.time()` timezone
+   
+ 
+|Type |Code   |Value |
+|---|---|---|
+|Hour |%H | Hour (0-24) |
+|Minute |%M | Minute (0-59) |
+|Second |%S | Second (0-59) |
+|Dy |%j | Day of year (1-366) |
+|Day |%d | Day of the month (1-31) |
+|Month |%m | Month (01-12) |
+|Month |%b | Month(abbreviated) |
+|Month |%B | Month(full name) |
+|Year |%y | Year (2 digit) |
+|Year |%Y | Year (4 digit) |
+
+Let's explore the following:
+```{r, results='hide'}
+x <- as.POSIXct("2021-02-23 20:30:00")
+x
+class(x)
+str(x)
+attr(x, "tzone")
+
+y <- as.POSIXct("April 2, 2019 15:35", format = "%B %d, %Y %H:%M",
+                tz = "America/Denver")
+y
+attr(y, "tzone")
+
+z <- ISOdatetime(year = 2021, month = 2, day = 23, hour = 20, min = 30, sec = 0,
+                 tz = "America/Phoenix")
+z
+
+# Try converting the following: 
+"06:59 Sep 02 1995"
+"16/Oct/2005:07:51:00"
+"12-Mar-2012 0830"
+2013; 8; 18
```

Comment:
What is the purpose of this example? Is it to add a time where there isn't yet one?

Reply:
These are intended to be hands-on exercises. The last case is if the date is provided as separate numeric values of year, month, and day.